PetApiTest/UnitTest1.cs

```diff
@@ -1,13 +1,259 @@
+            TestServer server = new TestServer(new WebHostBuilder().UseStartup<Startup>());
+            HttpClient client = server.CreateClient();
+            await client.DeleteAsync("PetStore/Clear");
+
+            Pet pet = new Pet("huahua", "dog", "black", 5000);
+            string request = JsonConvert.SerializeObject(pet);
+            StringContent requestBody = new StringContent(request, Encoding.UTF8, "application/json");
+
+            //when
+            await client.PostAsync("PetStore/addNewPet", requestBody);
+            var response = await client.GetAsync("PetStore/Pets");
+
+            //then
+            response.EnsureSuccessStatusCode();
+            var responseString = await response.Content.ReadAsStringAsync();
+            List<Pet> actrualPet = JsonConvert.DeserializeObject<List<Pet>>(responseString);
+            Assert.Equal(new List<Pet> { pet }, actrualPet);
+        }
+
+        [Fact]
+        public async Task Should_Return_Specific_Pet_When_Providing_Its_Name()
+        {
+            //given
+            TestServer server = new TestServer(new WebHostBuilder().UseStartup<Startup>());
+            HttpClient client = server.CreateClient();
+            await client.DeleteAsync("PetStore/Clear");
+
+            Pet pet = new Pet("huahua", "dog", "black", 5000);
+            string request = JsonConvert.SerializeObject(pet);
+            StringContent requestBody = new StringContent(request, Encoding.UTF8, "application/json");
+            await client.PostAsync("PetStore/addNewPet", requestBody);
+
+            //when
+            var response = await client.GetAsync($"PetStore/GetByName?name={pet.Name}");
+
+            //then
+            response.EnsureSuccessStatusCode();
+            var responseString = await response.Content.ReadAsStringAsync();
+            Pet actrualPet = JsonConvert.DeserializeObject<Pet>(responseString);
+            Assert.Equal(pet, actrualPet);
+        }
+
+        [Fact]
+        public async Task Should_Delete_Specific_Pet_When_Buy_Name()
+        {
+            //given
+            TestServer server = new TestServer(new WebHostBuilder().UseStartup<Startup>());
+            HttpClient client = server.CreateClient();
+            await client.DeleteAsync("PetStore/Clear");
+
+            Pet pet = new Pet("huahua", "dog", "black", 5000);
+            string request = JsonConvert.SerializeObject(pet);
+            StringContent requestBody = new StringContent(request, Encoding.UTF8, "application/json");
+            await client.PostAsync("PetStore/addNewPet", requestBody);
+
+            //when
+            var response = await client.DeleteAsync($"PetStore/DeleteByName?name={pet.Name}");
+
+            //then
+            response.EnsureSuccessStatusCode();
+            var responseString = await response.Content.ReadAsStringAsync();
+            List<Pet> actrualPet = JsonConvert.DeserializeObject<List<Pet>>(responseString);
+            Assert.Null(actrualPet);
```

Comment:
maybe use assert.comtains first to indicate the list is not null, then test the delete method